beacon-chain/state/state-native/readonly_validator.go

```diff
@@ -78,6 +79,28 @@ func (v readOnlyValidator) WithdrawalCredentials() []byte {
 	return creds
 }
 
+// HasETH1WithdrawalCredential returns whether the validator has an ETH1
+// Withdrawal prefix.
+func (v readOnlyValidator) HasETH1WithdrawalCredential() bool {
+	cred := v.WithdrawalCredentials()
+	return cred[0] == params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
```

Comment:
```suggestion
	return len(cred) > 0 && cred[0] == params.BeaconConfig().ETH1AddressWithdrawalPrefixByte
```

Reply:
Why this check? A validator can never have empty withrawal credentials, it's unmarshalled as 32 bytes

Reply:
It's a free defensive check and will make fuzzing work better.

Reply:
The risk of not having this is that the node will panic in the (presumably) impossible scenario where the credential slice is empty